target/linux/ramips/patches-4.4/0701-mt7620-add-mib-stats.patch

```diff
@@ -0,0 +1,280 @@
+Index: linux-4.4.27/drivers/net/ethernet/mediatek/gsw_mt7620.c
+===================================================================
+--- linux-4.4.27.orig/drivers/net/ethernet/mediatek/gsw_mt7620.c
++++ linux-4.4.27/drivers/net/ethernet/mediatek/gsw_mt7620.c
+@@ -67,6 +67,9 @@ static void mt7620_hw_init(struct mt7620
+ 	rt_sysc_w32(rt_sysc_r32(SYSC_REG_CFG1) | BIT(8), SYSC_REG_CFG1);
+ 	mtk_switch_w32(gsw, mtk_switch_r32(gsw, GSW_REG_CKGCR) & ~(0x3 << 4), GSW_REG_CKGCR);
+ 
++	// Enable MIB stats
++	mtk_switch_w32(gsw, mtk_switch_r32(gsw, GSW_REG_MIB_CNT_EN) | (1 << 1), GSW_REG_MIB_CNT_EN);
++
+ 	if (of_property_read_bool(np, "mediatek,mt7530")) {
+ 		u32 val;
+ 
+Index: linux-4.4.27/drivers/net/ethernet/mediatek/gsw_mt7620.h
+===================================================================
+--- linux-4.4.27.orig/drivers/net/ethernet/mediatek/gsw_mt7620.h
++++ linux-4.4.27/drivers/net/ethernet/mediatek/gsw_mt7620.h
+@@ -76,6 +76,8 @@
+ #define PHY_PRE_EN		BIT(30)
+ #define PMY_MDC_CONF(_x)	((_x & 0x3f) << 24)
+ 
++#define	GSW_REG_MIB_CNT_EN	0x4000
++
+ enum {
+ 	/* Global attributes. */
+ 	GSW_ATTR_ENABLE_VLAN,
+Index: linux-4.4.27/drivers/net/ethernet/mediatek/mt7530.c
+===================================================================
+--- linux-4.4.27.orig/drivers/net/ethernet/mediatek/mt7530.c
++++ linux-4.4.27/drivers/net/ethernet/mediatek/mt7530.c
+@@ -10,6 +10,7 @@
+  * GNU General Public License for more details.
+  *
+  * Copyright (C) 2013 John Crispin <blogic@openwrt.org>
++ * Copyright (C) 2016 Vitaly Chekryzhev <13hakta@gmail.com>
+  */
+ 
+ #include <linux/if.h>
+@@ -76,6 +77,95 @@ struct mt7xxx_mib_desc {
+ 	const char *name;
+ };
+ 
++#define MT7620_MIB_COUNTER_BASE	0x1010
```

Comment:
defines go in the header file
dont use comments at the end of a line
dont use c++ comments